core/src/test/java/org/apache/iceberg/TestTableMetadata.java

```diff
@@ -446,68 +448,48 @@ public void testVersionValidation() {
-
-      generator.writeStartObject(); // start table metadata object
-
-      generator.writeNumberField(FORMAT_VERSION, version);
-      generator.writeStringField(TABLE_UUID, metadata.uuid());
-      generator.writeStringField(LOCATION, metadata.location());
-      generator.writeNumberField(LAST_UPDATED_MILLIS, metadata.lastUpdatedMillis());
-      if (version > 1) {
-        generator.writeNumberField(TableMetadataParser.LAST_SEQUENCE_NUMBER, metadata.lastSequenceNumber());
-      }
-      generator.writeNumberField(LAST_COLUMN_ID, metadata.lastColumnId());
 
-      generator.writeFieldName(SCHEMA);
-      SchemaParser.toJson(metadata.schema(), generator);
-
-      // mimic an old writer by writing only partition-spec and not the default ID or spec list
-      generator.writeFieldName(PARTITION_SPEC);
-      PartitionSpecParser.toJsonFields(metadata.spec(), generator);
-
-      generator.writeObjectFieldStart(PROPERTIES);
-      for (Map.Entry<String, String> keyValue : metadata.properties().entrySet()) {
-        generator.writeStringField(keyValue.getKey(), keyValue.getValue());
-      }
-      generator.writeEndObject();
-
-      generator.writeNumberField(CURRENT_SNAPSHOT_ID,
-          metadata.currentSnapshot() != null ? metadata.currentSnapshot().snapshotId() : -1);
-
-      generator.writeArrayFieldStart(SNAPSHOTS);
-      for (Snapshot snapshot : metadata.snapshots()) {
-        SnapshotParser.toJson(snapshot, generator);
-      }
-      generator.writeEndArray();
-      // skip the snapshot log
+  @Test
+  public void testParserV2PartitionSpecsValidation() throws Exception {
+    String unsupportedVersion = readTableMetadataInputFile("TableMetadataV2MissingPartitionSpecs.json");
+    AssertHelpers.assertThrows("Should reject v2 metadata without partition specs",
+        IllegalArgumentException.class, "partition-specs must exist in format v2",
+        () -> TableMetadataParser.fromJson(
+            ops.io(), null, JsonUtil.mapper().readValue(unsupportedVersion, JsonNode.class))
+    );
+  }
 
-      generator.writeEndObject(); // end table metadata object
+  @Test
+  public void testParserV2SortOrderValidation() throws Exception {
+    String unsupportedVersion = readTableMetadataInputFile("TableMetadataV2MissingSortOrder.json");
+    AssertHelpers.assertThrows("Should reject v2 metadata without sort order",
+        IllegalArgumentException.class, "sort-orders must exist in format v2",
+        () -> TableMetadataParser.fromJson(
+            ops.io(), null, JsonUtil.mapper().readValue(unsupportedVersion, JsonNode.class))
+    );
+  }
 
-      generator.flush();
-    } catch (IOException e) {
-      throw new UncheckedIOException(String.format("Failed to write json for: %s", metadata), e);
-    }
-    return writer.toString();
+  private String readTableMetadataInputFile(String fileName) throws IOException {
+    Path path = Paths.get(new File("").getAbsolutePath(), "src/test/testfiles", fileName);
```

Comment:
I think the test files should be in `src/test/resources` so that `ClassLoader.getResource` can be used to read. That is usually a better way to find fixtures because it avoids assuming the current working directory of the build.

Reply:
Thank you for the review! Using `ClassLoader.getResource` is definitely cleaner. Updated.